src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public AbstractFigure getRandomFigure() {
+        Random random = new Random();
+        ColorSupplier colorSupplier = new ColorSupplier();
+        String color = colorSupplier.getRandomColor();
+        int figureType = random.nextInt(4);
```

Comment:
The `random.nextInt(4)` call will only generate numbers from 0 to 3. Therefore, the case `4` for `Rectangle` is unreachable. You should change `random.nextInt(4)` to `random.nextInt(5)` to include all cases.